### ISSUE TITLE
[RLlib] Memory overhead when creating RemoteEnv Actor

### ISSUE THREAD

**Issue by azzeddineCH** [ray-project/ray#34032]
### What happened + What you expected to happen

I’m trying to set up a training pipeline using ray remote env where I am creating 2 envs per CPU. after checking the ray dashboard. each remote env actor is taking up 600MB which relatively high given how simple is the dummy env I’m using on the example below .

`
class CustomEnv(gym.Env):

    def __init__(self, config):
        pass

    def reset(self):
        return spaces.Box(low=0, high=255, shape=(1,), dtype=np.uint8).sample()  # return random observation

    def step(self, action):
        obs = spaces.Box(low=0, high=255, shape=(1,), dtype=np.uint8).sample()  # return random observation
        reward = 0  # no reward
        return obs, reward, False, {}

    def observation_space(self):
        return spaces.Box(low=0, high=255, shape=(1,), dtype=np.uint8)

    def action_space(self):
        return spaces.Discrete(1)

    def _get_spaces(self):
        return spaces.Box(low=0, high=255, shape=(1,), dtype=np.uint8), spaces.Discrete(1)

def env_factory(env_config):
    return CustomEnv(env_config)

if __name__ == '__main__':
    # Register the environment with RLlib
    register_env("custom_env", env_factory)

    # Define the RLlib training configuration
    config = {
        "env": "custom_env",
        "num_workers": 1,
        "framework": "tf",
        "model": {
            "fcnet_hiddens": [256, 256]
        },
        "gamma": 0.99,
        "lr": 0.0005,
        "train_batch_size": 2000,
        "rollout_fragment_length": 100,
        "batch_mode": "truncate_episodes",
        "num_envs_per_worker": 2,
        "remote_worker_envs": True,
        "num_gpus": 0,
    }

    # Initialize the Ray cluster
    ray.init()

    # Train the RLlib agent
    analysis = tune.run(
        "PPO",
        config=config,
        stop={"episode_reward_mean": 200},
        checkpoint_at_end=True
    )

    # Print the results of the training
    print(analysis.dataframe())`

the memory taken by each remote env is shown on the ray dashboard (500MB which can't be explained): 

![Screen Shot 2023-04-04 at 01 06 08](https://user-images.githubusercontent.com/19895372/229653489-8892057b-605d-43f8-a094-14dc0e5f4677.png)

### Versions / Dependencies

python version: 3.8
ray version: 2.2.0
OS: macOS Big Sur 11.7.2

### Reproduction script

`class CustomEnv(gym.Env):

    def __init__(self, config):
        pass

    def reset(self):
        return spaces.Box(low=0, high=255, shape=(1,), dtype=np.uint8).sample()  # return random observation

    def step(self, action):
        obs = spaces.Box(low=0, high=255, shape=(1,), dtype=np.uint8).sample()  # return random observation
        reward = 0  # no reward
        return obs, reward, False, {}

    def observation_space(self):
        return spaces.Box(low=0, high=255, shape=(1,), dtype=np.uint8)

    def action_space(self):
        return spaces.Discrete(1)

    def _get_spaces(self):
        return spaces.Box(low=0, high=255, shape=(1,), dtype=np.uint8), spaces.Discrete(1)

def env_factory(env_config):
    return CustomEnv(env_config)

if __name__ == '__main__':
    # Register the environment with RLlib
    register_env("custom_env", env_factory)

    # Define the RLlib training configuration
    config = {
        "env": "custom_env",
        "num_workers": 1,
        "framework": "tf",
        "model": {
            "fcnet_hiddens": [256, 256]
        },
        "gamma": 0.99,
        "lr": 0.0005,
        "train_batch_size": 2000,
        "rollout_fragment_length": 100,
        "batch_mode": "truncate_episodes",
        "num_envs_per_worker": 2,
        "remote_worker_envs": True,
        "num_gpus": 0,
    }

    # Initialize the Ray cluster
    ray.init()

    # Train the RLlib agent
    analysis = tune.run(
        "PPO",
        config=config,
        stop={"episode_reward_mean": 200},
        checkpoint_at_end=True
    )

    # Print the results of the training
    print(analysis.dataframe())
`

### Issue Severity

None

**Comment by azzeddineCH** [comment]
it looks like this overhead comes from the fact that there is an import for both tensorflow and pytorch.

![Screen Shot 2023-04-04 at 13 59 28](https://user-images.githubusercontent.com/19895372/229799336-a2fdc59f-1497-4a6d-b3da-779cad123b4a.png)

**Comment by azzeddineCH** [comment]
The issue was mainly to the base env class importing some constants from the rllib.utils library which happened to have a global TF nad Pytorch import. I fixed this issue by patching the rllib env base class and deleting those imports.

**Comment by azzeddineCH** [comment]
this problem comes from the fact that I am starting a remote env process where my env is using the rllib BaseEnv. 
the rllib BaseEnv contains imports from the rllib.utils package (like annotation for PublicAPI, deprecated ... etc) cause that utils init package contains an import for tensorflow and pytorch on the head of the file 😄 .

**Comment by Rohan138** [comment]
The imports are all of the form `try_import_tf`, `try_import_torch`, so this shouldn't actually occur unless your remote envs are also have torch/tf installed in the runtime. Could you try moving the environment to a different file and setting `num_workers=0` to run the training on the local worker?

**Comment by Rohan138** [comment]
@azzeddineCH closing this issue, but please feel free to reopen if this bug reappears and your fix does not work. Thanks!
